Flaky e2e test TestTraceflowIntraNode

**Describe the bug**
TestTraceflowIntraNode sometimes failed.
```
2020-10-23T11:43:02.3783380Z === RUN   TestTraceflowIntraNode
2020-10-23T11:43:02.3784778Z ##[error]    fixtures.go:107: Creating 'antrea-test' K8s Namespace
2020-10-23T11:43:02.3786284Z ##[error]    fixtures.go:70: Applying Antrea YAML
2020-10-23T11:43:03.3002000Z ##[error]    fixtures.go:74: Waiting for all Antrea DaemonSet Pods
2020-10-23T11:43:04.3132514Z ##[error]    fixtures.go:78: Checking CoreDNS deployment
2020-10-23T11:43:35.1726552Z ##[error]Copying file antrea-agent.cov.out from pod antrea-agent-4khmg podName to '/home/runner/work/antrea/antrea/test-e2e-encap-coverage'Copying file antrea-agent.cov.out from pod antrea-agent-zd9fr podName to '/home/runner/work/antrea/antrea/test-e2e-encap-coverage'Copying file antrea-agent.cov.out from pod antrea-agent-zdvnw podName to '/home/runner/work/antrea/antrea/test-e2e-encap-coverage'Copying file antrea-controller.cov.out from pod antrea-controller-ff9f77466-5wkbm podName to '/home/runner/work/antrea/antrea/test-e2e-encap-coverage'    fixtures.go:318: Creating a busybox test Pod 'test-pod-hwjai1ti' and waiting for IP
2020-10-23T11:43:35.1732086Z ##[error]    fixtures.go:318: Creating a busybox test Pod 'test-pod-mwo99x75' and waiting for IP
2020-10-23T11:43:35.1734083Z ##[error]    fixtures.go:318: Creating a busybox test Pod 'test-pod-0xrovi0w' and waiting for IP
2020-10-23T11:43:37.8755315Z === RUN   TestTraceflowIntraNode/traceflowGroupTest
2020-10-23T11:43:37.8756513Z === RUN   TestTraceflowIntraNode/traceflowGroupTest/intraNodeTraceflow
2020-10-23T11:43:38.9084283Z ##[error]    traceflow_test.go:790: Observations should be [{SpoofGuard  Forwarded    0   } {NetworkPolicy EgressRule Forwarded    0   } {NetworkPolicy IngressDefaultRule Dropped    0   }], but got [{SpoofGuard  Forwarded    0   } {Forwarding Output Delivered    0   }]
2020-10-23T11:43:38.9184677Z === RUN   TestTraceflowIntraNode/traceflowGroupTest/intraNodeUDPDstPodTraceflow
2020-10-23T11:43:39.9513324Z === RUN   TestTraceflowIntraNode/traceflowGroupTest/intraNodeUDPDstIPTraceflow
2020-10-23T11:43:40.9902242Z === RUN   TestTraceflowIntraNode/traceflowGroupTest/intraNodeICMPDstIPTraceflow
2020-10-23T11:43:42.0169355Z === RUN   TestTraceflowIntraNode/traceflowGroupTest/nonExistingDstPod
2020-10-23T11:43:43.0672578Z === CONT  TestTraceflowIntraNode
2020-10-23T11:43:43.0680644Z ##[error]    fixtures.go:281: Deleting Pod 'test-pod-hwjai1ti'
2020-10-23T11:43:43.0682396Z ##[error]    fixtures.go:281: Deleting Pod 'test-pod-mwo99x75'
2020-10-23T11:43:43.0684151Z ##[error]    fixtures.go:281: Deleting Pod 'test-pod-0xrovi0w'
2020-10-23T11:43:43.1017753Z ##[error]    fixtures.go:167: Exporting test logs to '/home/runner/work/antrea/antrea/log/TestTraceflowIntraNode/beforeTeardown.Oct23-11-43-43'
2020-10-23T11:43:45.6056271Z ##[error]    fixtures.go:274: Deleting 'antrea-test' K8s Namespace
2020-10-23T11:43:54.6152364Z --- FAIL: TestTraceflowIntraNode (52.24s)
2020-10-23T11:43:54.6153274Z     --- FAIL: TestTraceflowIntraNode/traceflowGroupTest (5.17s)
2020-10-23T11:43:54.6154296Z         --- FAIL: TestTraceflowIntraNode/traceflowGroupTest/intraNodeTraceflow (1.04s)
2020-10-23T11:43:54.6155557Z         --- PASS: TestTraceflowIntraNode/traceflowGroupTest/intraNodeUDPDstPodTraceflow (1.03s)
2020-10-23T11:43:54.6156896Z         --- PASS: TestTraceflowIntraNode/traceflowGroupTest/intraNodeUDPDstIPTraceflow (1.04s)
2020-10-23T11:43:54.6158230Z         --- PASS: TestTraceflowIntraNode/traceflowGroupTest/intraNodeICMPDstIPTraceflow (1.03s)
2020-10-23T11:43:54.6159452Z         --- PASS: TestTraceflowIntraNode/traceflowGroupTest/nonExistingDstPod (1.03s)
```
It's because the way it checks whether a networkpolicy is realized is not valid now after `antctl get networkpolicy NAME -n NAMESPACE` can return multiple objects (it no longer returns error code for NotFound case).
[e2e-kind-encap.tar.gz (4).zip](https://github.com/vmware-tanzu/antrea/files/5456542/e2e-kind-encap.tar.gz.4.zip)


**To Reproduce**
Randomly failing.

**Expected**
The tests that check whether a networkpolicy is realized on a Node should not rely on exit code.

**Versions:**
Please provide the following information:
 - Antrea version (Docker image tag). 0.11.0-dev (master)

--- a/test/e2e/traceflow_test.go
+++ b/test/e2e/traceflow_test.go
@@ -26,6 +26,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 
+	"github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta2"
 	"github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
 )
 
@@ -746,18 +747,16 @@ func (data *TestData) createNPAllowAllEgress(name string) (*networkingv1.Network
 // waitForNetworkpolicyRealized waits for the NetworkPolicy to be realized by the antrea-agent Pod.
 func (data *TestData) waitForNetworkpolicyRealized(pod string, networkpolicy string) error {
 	if err := wait.Poll(200*time.Millisecond, 5*time.Second, func() (bool, error) {
-		cmds := []string{"antctl", "get", "networkpolicy", "-S", networkpolicy, "-n", testNamespace}
-		if _, stderr, err := runAntctl(pod, cmds, data); err != nil {
-			if strings.Contains(stderr, "server could not find the requested resource") {
-				return false, nil
-			}
-			return false, err
+		cmds := []string{"antctl", "get", "networkpolicy", "-S", networkpolicy, "-n", testNamespace, "-T", "K8sNP"}
+		stdout, stderr, err := runAntctl(pod, cmds, data)
+		if err != nil {
+			return false, fmt.Errorf("Error when executing antctl get NetworkPolicy, stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
 		}
-		return true, nil
+		return strings.Contains(stdout, fmt.Sprintf("%s:%s/%s", v1beta2.K8sNetworkPolicy, testNamespace, networkpolicy)), nil
 	}); err == wait.ErrWaitTimeout {
 		return fmt.Errorf("NetworkPolicy %s isn't realized in time", networkpolicy)
 	} else if err != nil {
-		return fmt.Errorf("Error when executing antctl get NetworkPolicy: %v", err)
+		return err
 	}
 	return nil
 }

